### ISSUE TITLE
Add language extension code action targets the nearest {-# pragma

### ISSUE THREAD

**Issue by isovector** [haskell/haskell-language-server#1958]
Given this:

```haskell
data Metaprogram = Metaprogram
  { mp_name             :: !Text
  , mp_known_by_auto    :: !Bool
  , mp_show_code_action :: !Bool
  , mp_program          :: !(TacticsM ())
  }
  deriving stock Generic
{-# ANN Metaprogram "hello" #-}

instance NFData Metaprogram where
  rnf (!(Metaprogram !_ !_ !_ !_)) = ()
```

without `-XBangPatterns` enabled, I get a code action on the last line, suggesting I add `BangPatterns`. Doing so produces this output:

```haskell
data Metaprogram = Metaprogram
  { mp_name             :: !Text
  , mp_known_by_auto    :: !Bool
  , mp_show_code_action :: !Bool
  , mp_program          :: !(TacticsM ())
  }
  deriving stock Generic
{-# ANN Metaprogram "hello" #-}
{-# LANGUAGE BangPatterns #-}

instance NFData Metaprogram where
  rnf (!(Metaprogram !_ !_ !_ !_)) = ()
```

which is super the wrong place for it.

**Comment by isovector** [comment]
Also happens with `INLINE` pragmas

**Comment by Ailrun** [comment]
https://github.com/haskell/haskell-language-server/blob/d4e2a6f6af9bc2536f6b3208f236546a74484004/plugins/hls-pragmas-plugin/src/Ide/Plugin/Pragmas.hs#L192
is the line caused the problem.

**Comment by jneira** [comment]
pragmas plugin is having a hard time finding the correct place to add the extension :smile: 
we must add a regression test for this (and over whatever corner case we can think of!)

**Comment by jneira** [comment]
This was closed by #2043 i suppose, closing optimistically
